hastexo/hastexo.py

```diff
@@ -992,10 +999,37 @@ def set_score(self, score):
             'raw_possible': score.raw_possible
         }
 
-    def publish_grade(self, score):
+    def get_score(self):
+        """
+        Return a raw score already persisted on the XBlock.  Should not
+        perform new calculations.
+        Returns:
+            Score(raw_earned=float, raw_possible=float)
+        """
+        if not self.score:
+            logger.warning("No score is earned for this block yet")
```

Comment:
Just checking: is this really a "warning" condition? Wouldn't we call this every time an instructor downloads a grade report, rather needlessly flooding the logs? (Consider what happens in a course with 1,000 enrolled learners.)

Reply:
Good question, I am actually not sure how to download a grade report. I believe the grades are generally fetched from the database so as far as I know we are only calling this after a score is set from a student's submission or from instructor's override, so we really shouldn't see this warning at all, but I added it for future us to help tracing down some unexpected behaviour. I really don't know at this point if we should be returning None or maybe a default 0/max_score here.